STEP_Project/OUListContent.js

```diff
@@ -4,6 +4,80 @@ if (chrome.storage !== undefined)
     storageObj = chrome.storage.sync;
 }
 
+/**
+ * This function adds the selected entity to Chrome's storage
+ * @param {HTML tr object} row Each row in the table of entities
+ * @param {string} dataid The data-row-id for each entity
+ */
+function addEntity(row, dataid)
+{
+    event.target.innerHTML = '-';
+    event.target.setAttribute('class','mClass'); 
+    let dataname = ''; 
+    var key = dataid; 
+
+    //if DOM structure is not as expected
+    if (row === undefined || row.firstChild === undefined)
+    {
+        alert('error in DOM');
+    }
+
+    //if entity is a group
+    else if (row.getAttribute('data-group-name') !== null)
+    {
+        dataname = row.getAttribute('data-group-name');
+        key = 'group-' + dataid;
+    }
+
+    //if entitiy is a user
+    else if (row.getAttribute('data-url') !== null)
+    {
+        dataname = (row.children[1].children[1].children[0].children[1].children[0].getAttribute('title'));
+        key = 'user-' + dataid;
+    }
+
+    //if entity is an OU
+    else if (row.children[0].children[0] !== undefined) //if there is an entity tree in the DOM
+    {
+        dataname = row.children[0].children[0].children[0].children[1].innerHTML;
```

Comment:
(optional) You can use firstChild in place of children[0]

Reply:
Replaced with firstChild